mp4parse/src/boxes.rs

```diff
@@ -94,6 +94,11 @@ impl fmt::Display for FourCC {
 
 box_database!(
     FileTypeBox                       0x6674_7970, // "ftyp"
+    MediaDataBox                      0x6d64_6174, // "mdat"
+    PrimaryItemBox                    0x7069_746d, // "pitm"
+    ItemInfoBox                       0x6969_6e66, // "iinf"
+    ItemInfoEntry                     0x696e_6665, // "infe"
```

Comment:
I was matching the names used in 14496-12:2015 § 8.11.6.2 (the syntax section):
![Screen Shot 2020-02-26 at 12 43 13 PM](https://user-images.githubusercontent.com/5906042/75386066-ded39100-5895-11ea-99fb-f9c5c6faf672.png)
![Screen Shot 2020-02-26 at 12 43 02 PM](https://user-images.githubusercontent.com/5906042/75386068-df6c2780-5895-11ea-8c5d-33d10d44fa5c.png)

`iinf` doesn't have it's own section, so the only reference to its full name is the syntax section for `infe`, which uses `ItemInfoBox`. The name `ItemInformationBox` doesn't appear in the spec.

This is contrary to the existing code's [`minf`](https://github.com/mozilla/mp4parse-rust/blob/bbb4a1217d2826e95b0a843e73c9e1a2faa23953/mp4parse/src/boxes.rs#L106) and [`schi`](https://github.com/mozilla/mp4parse-rust/blob/bbb4a1217d2826e95b0a843e73c9e1a2faa23953/mp4parse/src/boxes.rs#L136) whose names in the section heading and syntax section both use `Information` rather than `Info`:
![Screen Shot 2020-02-26 at 12 47 53 PM](https://user-images.githubusercontent.com/5906042/75386752-11ca5480-5897-11ea-94e3-65e5f3c432ff.png)
![Screen Shot 2020-02-26 at 12 48 16 PM](https://user-images.githubusercontent.com/5906042/75386795-260e5180-5897-11ea-8d97-9ee660e3d28b.png)

[`sinf`](https://github.com/mozilla/mp4parse-rust/blob/bbb4a1217d2826e95b0a843e73c9e1a2faa23953/mp4parse/src/boxes.rs#L138), on the other hand uses "Protection Scheme Information Box" as the title of § 8.12.1, but `ProtectionSchemeInfoBox` in the syntax section:
![Screen Shot 2020-02-26 at 12 48 38 PM](https://user-images.githubusercontent.com/5906042/75386828-358d9a80-5897-11ea-874b-9a4cab0dc470.png)

It's frustrating that the spec isn't more internally consistent, but which would we rather use as normative? And in the case that a box name is only present in the syntax section, what then?

Related: I'm not sure where the [`itif`](https://github.com/mozilla/mp4parse-rust/blob/bbb4a1217d2826e95b0a843e73c9e1a2faa23953/mp4parse/src/boxes.rs#L152) definition comes from. I don't see it in any of the ISO specs I have, nor does any of the other code seem to reference `itif` or `MetadataItemInformationBox`. Is it necessary?

Reply:
My mistake, let's stick with the names you've got.  I was looking at the section titles and assuming the spec was consistent between those and the syntax sections.  We should be using the names from the syntax sections where we can.

Looks like `itif` is snuck in with the iTunes metadata patches but is completely unused, we should remove it.  Good catch.